src/context/TransactionsProvider/TransactionsProvider.tsx

```diff
@@ -1,165 +1,105 @@
+import { useAppDispatch } from 'state/store'
+
+import { usePlugins } from '../PluginProvider/PluginProvider'
 
 const moduleLogger = logger.child({ namespace: ['TransactionsProvider'] })
 
 type TransactionsProviderProps = {
   children: React.ReactNode
 }
 
-export const TransactionsProvider = ({ children }: TransactionsProviderProps): JSX.Element => {
-  const dispatch = useDispatch()
+export const TransactionsProvider: React.FC<TransactionsProviderProps> = ({ children }) => {
+  const dispatch = useAppDispatch()
+  const [isSubscribed, setIsSubscribed] = useState<boolean>(false)
   const {
-    state: { wallet, walletInfo },
+    state: { wallet },
   } = useWallet()
-  const { supportedChains } = usePlugins()
-  const [isSubscribed, setIsSubscribed] = useState<boolean>(false)
-  const assets = useSelector(selectAssets)
-  const portfolioAssetIds = useSelector(selectPortfolioAssetIds)
-  const accountSpecifiers = useSelector(selectAccountSpecifiers)
+  const portfolioAccountMetadata = useSelector(selectPortfolioAccountMetadata)
   const portfolioLoadingStatus = useSelector(selectPortfolioLoadingStatus)
-  const txHistoryStatus = useAppSelector(selectTxHistoryStatus)
+  const { supportedChains } = usePlugins()
 
-  const chainAdapterManager = getChainAdapterManager()
   /**
    * unsubscribe and cleanup logic
    */
   useEffect(() => {
-    // account specifiers changing will trigger this effect
     // we've disconnected/switched a wallet, unsubscribe from tx history and clear tx history
     if (!isSubscribed) return
-    moduleLogger.debug('unsubscribing txs')
-    supportedChains.forEach(chainId => chainAdapterManager.get(chainId)?.unsubscribeTxs())
+    moduleLogger.debug({ supportedChains }, 'unsubscribing txs')
+    // this is heavy handed but will ensure we're unsubscribed from everything
+    supportedChains.forEach(chainId => getChainAdapterManager().get(chainId)?.unsubscribeTxs())
     setIsSubscribed(false)
-    // setting isSubscribed to false will trigger this effect
+    // isSubscribed causes spurious unsubscriptions - this will react correctly when portfolioAccountMetadata changes
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [accountSpecifiers, dispatch, chainAdapterManager, supportedChains])
+  }, [supportedChains, wallet, portfolioAccountMetadata])
 
   /**
    * tx history subscription logic
    */
   useEffect(() => {
+    if (isSubscribed) return
     if (!wallet) return
-    if (isEmpty(assets)) return
+    const accountIds = Object.keys(portfolioAccountMetadata)
+    if (!accountIds.length) return
+    // this looks useless, but prevents attempting to subscribe multiple times
+    // something further up the tree from this provider is causing renders when the portfolio status changes,
+    // even though it shouldn't
     if (portfolioLoadingStatus === 'loading') return
-    if (isSubscribed) return // don't resubscribe
-    if (txHistoryStatus !== 'loaded') return
```

Comment:
i feel like this was related to the event loop issues we saw a while back and may no longer be a problem, but just want to note that there was intention to wait until we fetched all txhistory before subscribing and consuming a chunk of request connections for the websocket subscriptions

Reply:
yeah, that indeed was, but we're definitely not seeing those problems now